### PR TITLE
Add raw_data parameter to SeldonClient predict

### LINKED ISSUE #3079: Add raw_data parameter to predict / transform / etc functions in Seldon Client
Recently a raw_data parameter was added to the `feedback` function of the SeldonClient module which allows for users to provide a raw dictionary representing the json request or the raw grpc proto which would then be sent directly as a request to the deployed Seldon model. This is a required feature for the batch processing component in case request-metadata is to be sent, and reduces the processing time hence making the requests more efficient.

### PR DESCRIPTION
**Pull request by nadinet**
**What this PR does / why we need it**:
Add raw_data parameter to SeldonClient predict. enable raw  payload for REST/gRpc. 
Add unit testing and upadate sklearn_iris notebook eample to include raw_data use example.
**Which issue(s) this PR fixes**:
https://github.com/SeldonIO/seldon-core/issues/3079

Fixes #3079 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note
```

### CODE REVIEW

File: examples/models/sklearn_iris/sklearn_iris.ipynb
Comment (RafalSkolasinski) [inline examples/models/sklearn_iris/sklearn_iris.ipynb:184]: we should not change port `8003` to `80` - most examples assumes that default gateway is on `8003`

**Review by RafalSkolasinski — CHANGES_REQUESTED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ryandawsonuk**
You can assign the PR to them by writing `/assign @ryandawsonuk` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **ryandawsonuk**
You can assign the PR to them by writing `/assign @ryandawsonuk` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: python/seldon_core/seldon_client.py
Comment (adriangonz) [inline python/seldon_core/seldon_client.py:1452]: Is the `request` variable needed here? It seems that after the big `if` / `else` only `payload` is used.

File: python/tests/test_seldon_client.py
Comment (adriangonz) [inline python/tests/test_seldon_client.py:56]: Won't the `json` variable match what's in `RAW_DATA_TEST`? Also, since `mocked_requests_post_success_raw_data` only seems to mock the response, can't we use one of the existing mock side effects?

File: python/tests/test_seldon_client.py
Comment (adriangonz) [inline python/tests/test_seldon_client.py:137]: Do we need assertions for the response values?

File: python/tests/test_seldon_client.py
Comment (adriangonz) [inline python/tests/test_seldon_client.py:337]: Interesting. Do you know where is this `strData` value coming from?

File: python/seldon_core/seldon_client.py
Comment (nadinet) [reply python/seldon_core/seldon_client.py:1452]: used also in line #1556 (return SeldonClientPrediction(request, response, success, msg))

File: python/tests/test_seldon_client.py
Comment (nadinet) [reply python/tests/test_seldon_client.py:56]: 1. regarding json variable: did same implementation as in mocked_requests_post_success_json_data. this is done multiple times in the client code also - I understand from Alejandro that it is a kind of typing checks / make sure... ( ofcource itis not so relevant in unit-testings)
2. mocked_requests_post_success_json_data can be use - but the return value will be of JSON_TEST_DATA. I defined a new one for competible.

File: python/tests/test_seldon_client.py
Comment (nadinet) [reply python/tests/test_seldon_client.py:137]: acctually I kept the same correction checks as in the other test_predict_rest_ .... methods. 
clearly it can be remove - but why it is there for the first place?

File: python/tests/test_seldon_client.py
Comment (nadinet) [reply python/tests/test_seldon_client.py:337]: Yepp - from the MyStub, line #279

File: examples/models/sklearn_iris/sklearn_iris.ipynb
Comment (nadinet) [reply examples/models/sklearn_iris/sklearn_iris.ipynb]: Done

File: python/tests/test_seldon_client.py
Comment (nadinet) [reply python/tests/test_seldon_client.py:56]: will change to use moked....post_success

File: python/tests/test_seldon_client.py
Comment (nadinet) [reply python/tests/test_seldon_client.py:337]: remove return value check

**Comment by axsaucedo** [comment]
Looks good
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3095#issuecomment-808391882 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
